src/services/drpcEvents/index.ts

```diff
@@ -206,15 +209,20 @@ export default class DrpcEvents {
       }
 
       //update corresponding query
-      const [query] = await this.db.update(
+      const [query]: QueryRow[] = await this.db.update(
         'query',
         { id: queryRow.id },
         { query_response: params.emissions, status: 'resolved' }
       )
 
+      if (queryRow.parent_id) {
+        await this.handleParentQuery(queryRow.parent_id, queryRow, params)
+      }
```

Comment:
I'm somewhat concerned by error handling here, but let's roll with it for now.

Reply:
mmm... thought we should handle and In my mind I've a vision of simplifying errors e.g. single try/catch or `if (this) doThat()` and break down into smaller functions to avoid nesting where we do database lookups/queries but as it does change the scope and handle throws instead.

So like in scope3co2 we return error html

```
      this.scope3CarbonConsumptionTemplates.newScope3CarbonConsumptionFormPage({
        formStage: 'error',
```

but instead we log and handle:
-  db update status -> error
- if we have drpc req.id -> return something like along with setting query to state error
```
      await this.cloudagent.submitDrpcResponse(request.id, {
        error: {
          code: drpcErrorCode.INTERNAL_ERROR, // not sure
          message: <error where we have failed>,
        },
      })
```

This also brings another question: how we would like to handle (partials) when one of the childs throws an error @ the parent level